ui/pages/api/stripe/create-checkout-session.ts

```diff
@@ -66,118 +69,165 @@ async function getTaxRates({
 }
 
 export default handler().post(async (req, res) => {
-  if (typeof req.query?.bucketId !== "string") throw new Error("Bad bucketId");
-  if (typeof req.query?.contribution !== "string")
-    throw new Error("Bad contribution");
-  if (typeof req.query?.tipAmount !== "string")
-    throw new Error("Bad tipAmount");
-  const bucketId = req.query?.bucketId;
-  const contribution = Number(req.query?.contribution);
-  const tipAmount = Number(req.query?.tipAmount);
-
-  // throws if not a round member
-  const roundMember = await getRoundMember({
-    bucketId,
-    userId: req.user.id,
-    include: { user: true },
-  });
-
-  const bucket = await prisma.bucket.findUnique({
-    where: { id: bucketId },
-    include: { round: { include: { group: true } } },
-  });
-
-  if (!bucket.directFundingEnabled || !bucket.round.directFundingEnabled) {
-    throw new Error("Direct funding not enabled for this bucket and/or round");
-  }
+  if (req.query?.mode === "paidplan") {
+    if (!req.user) throw new Error("You need to be logged in");
+    if (typeof req.query?.plan !== "string") throw new Error("No plan specified");
+    if (typeof req.query?.groupSlug !== "string") throw new Error("No plan specified");
+    if (typeof req.query?.groupName !== "string") throw new Error("No plan specified");
+
+    //if (typeof req.query?.contribution !== "string")
+
+    const priceId = plans[req.query.plan];
+    if (!priceId) throw new Error("Missing price ID for this plan.");
+
+    const customerMetadata = {
+      customer_email: req.user.email,
+    };
+    const origin = getRequestOrigin(req);
+    try {
+      const session = await stripe.checkout.sessions.create({
+        mode: "subscription",
+        payment_method_types: ["card"],
+        tax_id_collection: {
+          enabled: true,
+        },
+        line_items: [
+          {
+            price: priceId,
+            quantity: 1,
+          },
+        ],
+        metadata: {
+          userId: req.user.id,
+          groupSlug: req.query.groupSlug,
+          groupName: req.query.groupName,
+        },
+        allow_promotion_codes: true,
+        ...customerMetadata,
+        billing_address_collection: "auto",
+        success_url: `${origin}/new-group/?upgraded=true&group=${slugify(req.query.groupSlug)}`,
+        cancel_url: `${origin}/new-group/?upgraded=false`,
```

Comment:
I don't think we can redirect like this, it has to be a static url. See the example lower down in this file on how to do it instead

Reply:
Hm, it is working in development. Any reasons why it should not work?

Reply:
I think in prod they might be stricter. I'll try to find the docs for it